sentry/src/main/java/io/sentry/CombinedScopeView.java

```diff
@@ -0,0 +1,456 @@
+      return isolationScope;
+    }
+    return globalScope;
+  }
+
+  @Override
+  public @NotNull List<Attachment> getAttachments() {
+    final @NotNull List<Attachment> allAttachments = new CopyOnWriteArrayList<>();
+    allAttachments.addAll(globalScope.getAttachments());
+    allAttachments.addAll(isolationScope.getAttachments());
+    allAttachments.addAll(scope.getAttachments());
+    return allAttachments;
+  }
+
+  @Override
+  public void addAttachment(@NotNull Attachment attachment) {
+    getDefaultWriteScope().addAttachment(attachment);
+  }
+
+  @Override
+  public void clearAttachments() {
+    getDefaultWriteScope().clearAttachments();
+  }
+
+  @Override
+  public @NotNull List<EventProcessor> getEventProcessors() {
+    // TODO mechanism for ordering event processors
+    final @NotNull List<EventProcessor> allEventProcessors = new CopyOnWriteArrayList<>();
+    allEventProcessors.addAll(globalScope.getEventProcessors());
+    allEventProcessors.addAll(isolationScope.getEventProcessors());
+    allEventProcessors.addAll(scope.getEventProcessors());
+    return allEventProcessors;
+  }
+
+  @Override
+  public void addEventProcessor(@NotNull EventProcessor eventProcessor) {
+    getDefaultWriteScope().addEventProcessor(eventProcessor);
+  }
+
+  @Override
+  public @Nullable Session withSession(Scope.@NotNull IWithSession sessionCallback) {
+    return getDefaultWriteScope().withSession(sessionCallback);
+  }
+
+  @Override
+  public @Nullable Scope.SessionPair startSession() {
+    return getDefaultWriteScope().startSession();
+  }
+
+  @Override
+  public @Nullable Session endSession() {
+    return getDefaultWriteScope().endSession();
+  }
+
+  @Override
+  public void withTransaction(Scope.@NotNull IWithTransaction callback) {
+    getDefaultWriteScope().withTransaction(callback);
+  }
+
+  @Override
+  public @NotNull SentryOptions getOptions() {
+    return scope.getOptions();
+  }
+
+  @Override
+  public @Nullable Session getSession() {
+    final @Nullable Session current = scope.getSession();
+    if (current != null) {
+      return current;
+    }
+    final @Nullable Session isolation = isolationScope.getSession();
+    if (isolation != null) {
+      return isolation;
+    }
+    return globalScope.getSession();
+  }
+
+  @Override
+  public void setPropagationContext(@NotNull PropagationContext propagationContext) {
+    getDefaultWriteScope().setPropagationContext(propagationContext);
+  }
+
+  @Override
+  public @NotNull PropagationContext getPropagationContext() {
+    return getDefaultWriteScope().getPropagationContext();
+  }
+
+  @Override
+  public @NotNull PropagationContext withPropagationContext(
+      Scope.@NotNull IWithPropagationContext callback) {
+    return getDefaultWriteScope().withPropagationContext(callback);
+  }
+
+  @Override
+  public @NotNull IScope clone() {
+    // TODO just return a new CombinedScopeView with forked scope?
```

Comment:
Not quite sure here.

Reply:
We're now returning a new `CombinedScopeView` with forked isolation and current scope. See https://github.com/getsentry/sentry-java/pull/3375